src/uk/gov/hmcts/contino/GithubAPI.groovy

```diff
@@ -38,65 +110,113 @@ class GithubAPI {
    *   A List of labels
    */
   def addLabels(project, issueNumber, labels) {
-
+    this.steps.echo "Adding the following labels: ${labels}"
     def body = JsonOutput.toJson(labels)
-
+    this.steps.echo "Request Body: ${body}"
     def response = this.steps.httpRequest(httpMode: 'POST',
       authentication: this.steps.env.GIT_CREDENTIALS_ID,
       acceptType: 'APPLICATION_JSON',
       contentType: 'APPLICATION_JSON',
-      url: "${API_URL}/${project}/issues/${issueNumber}/labels",
+      url: API_URL + "/${project}/issues/${issueNumber}/labels",
       requestBody: "${body}",
       consoleLogResponseBody: true,
       validResponseCodes: '200')
-  }
 
-/**
- * Check Pull Request for label by a pattern in name.
- */
-  def getLabelsbyPattern(String branch_name, String key) {
-    if (new ProjectBranch(branch_name).isPR() == true) {
-      def project = currentProject()
-      def pullRequestNumber = currentPullRequestNumber()
-      return getLabels(project, pullRequestNumber).findAll{it.contains(key)}
+    this.steps.echo "Response Status Code: ${response.status}"
+
+    if (response.status == 200) {
+      this.steps.echo "Response Ok."
+      if (isCacheValid()) {
+        this.steps.echo "Cache is Valid.  Adding new labels to cache."
+        cachedLabelList.cache.addAll(labels)
+        cachedLabelList.cache.unique()
+        this.steps.echo "Updated Cache Contents: ${getCache()}"
+      } else {
+        this.steps.echo "Cache is Invalid.  Refreshing Cache."
```

Comment:
also do we need this?
```suggestion
        this.steps.echo "Cache is Invalid. Refreshing Cache."
```

Reply:
Removed the additional logging (not required outside of initial testing)